datafusion/core/tests/user_defined/user_defined_scalar_functions.rs

```diff
@@ -518,16 +514,12 @@ impl ScalarUDFImpl for AddIndexToStringVolatileScalarUDF {
         Ok(self.return_type.clone())
     }
 
-    fn invoke_batch(
-        &self,
-        args: &[ColumnarValue],
-        number_rows: usize,
-    ) -> Result<ColumnarValue> {
-        let answer = match &args[0] {
+    fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
+        let answer = match &args.args[0] {
```

Comment:
As a minor follow up, this could use `take_function_args`

https://github.com/apache/datafusion/blob/2d303346c7d44a78d317e7b7b959cbd94edfb891/datafusion/functions-aggregate/src/sum.rs#L130-L129